platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/LearningSynchronizer.java

```diff
@@ -122,6 +128,10 @@ public LearningSynchronizer(
         rootsToReceive.add(root);
 
         this.breakConnection = breakConnection;
+
+        this.mapStats = metrics == null // to support unit tests
```

Comment:
Why not just pass in a mockito object for unit tests when no metrics are needed? That's how this is handled elsewhere in the code.

Reply:
Because we currently don't have any tests for the metrics, and therefore this extra complexity is unnecessary today. But okay, let's mock them.